roles/kubernetes/preinstall/handlers/main.yml

```diff
@@ -120,3 +120,14 @@
   service:
     name: systemd-resolved
     state: restarted
+
+- name: Preinstall | restart ntp
+  service:
+    name: >-
+      {% if ntp_package == "ntp" -%}
```

Comment:
This pattern repeats itself across your patch, it's better to `set_fact` a private variable and reuse it instead of copying the conditional code across.